staging/src/k8s.io/legacy-cloud-providers/aws/aws_assumerole_provider.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+)
+
+const (
+	invalidateCredsAfter = 1 * time.Second
```

Comment:
Seems like your invalidating the cache rather than the credential...

Reply:
cache is holding the creds, so I wanted make it explicit what will be expiring :)

Reply:
Good point though, it could be more explicit like `invalidateCredsCacheAfter`

Reply:
Updated the name to `invalidateCredsCacheAfter`. PTAL.